src/plugins/usage_collection/server/collector/collector_set.ts

```diff
@@ -92,51 +96,72 @@ export class CollectorSet {
     return [...this.collectors.values()].find((c) => c.type === type);
   };
 
-  public areAllCollectorsReady = async (collectorSet: CollectorSet = this) => {
-    if (!(collectorSet instanceof CollectorSet)) {
+  private getReadyCollectors = async (
+    collectors: Map<string, AnyCollector> = this.collectors
+  ): Promise<AnyCollector[]> => {
+    if (!(collectors instanceof Map)) {
       throw new Error(
-        `areAllCollectorsReady method given bad collectorSet parameter: ` + typeof collectorSet
+        `areAllCollectorsReady method given bad Map of collectors: ` + typeof collectors
       );
     }
 
-    const collectors = [...collectorSet.collectors.values()];
-    const collectorsWithStatus = await Promise.all(
-      collectors.map(async (collector) => {
-        return {
-          isReady: await collector.isReady(),
-          collector,
-        };
+    const secondInMs = 1000;
+    const collectorsWithStatus: CollectorWithStatus[] = await Promise.all(
+      [...collectors.values()].map(async (collector) => {
+        const isReadyWithTimeout = await withTimeout<boolean>({
+          promise: (async (): Promise<boolean> => {
+            try {
+              return await collector.isReady();
+            } catch (err) {
+              this.logger.debug(
+                `Collector failed to get ready` +
+                  `${this.maximumWaitTimeForAllCollectorsInS}s and will return data from all collectors that are ready.`
```

Comment:
> and will return data from all collectors that are ready.

Does it make sense to say this for every collector that fails?

Reply:
True we already mentioned that in the next lines as well